### PR TITLE
Unify keyboard events on docs.rs results

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
The keyboard events are suboptimal currently. So I improved them a bit by following what @jsha did in rustdoc. It allows to also centralize the JS in one file and remove the inline JS we currently have. Small demo below:

https://user-images.githubusercontent.com/3050060/126352782-138424f0-15c8-4a76-a879-0a328cdfef27.mp4

### CODE REVIEW

**Comment by syphar** [comment]
@GuillaumeGomez I would prefer to leave the review with someone knowing more JS than I do, 

though if this takes too much time I would at least dig into it and validate it works as it should :)

**Comment by GuillaumeGomez** [comment]
It was more to get an "overall" review. There are DOM changes after all. ;)

File: templates/core/home.html [hunk 1]
```diff
@@ -32,70 +32,41 @@ <h1 class="brand">{{ "cubes" | fas(fw=true) }} Docs.rs</h1>
...
-                                {%- endif -%}
+                        {%- if varsb.show_stars -%}
+                            <div class="pure-u-1 pure-u-sm-4-24 pure-u-md-3-24 date">
+                                {{ release.stars }}
+                                {{ "star" | fas }}
                             </div>
-                        </a>
-                    </li>
-                {%- endfor -%}
-            </ul>
```
Comment (jyn514) [inline templates/core/home.html:69]: Why did you remove these HTML tags?

File: templates/releases/releases.html [hunk 2]
```diff
@@ -18,43 +18,38 @@
...
+                            </div>
+                        {%- else -%}
+                            <div class="pure-u-1 pure-u-sm-4-24 pure-u-md-3-24 date"
+                                title="{{ release.release_time | date(format='%FT%TZ') }}">
+                                {{ release.release_time | timeformat(relative=true) }}
                             </div>
-                        </a>
-                    </li>
-                {%- endfor -%}
-            </ul>
```
Comment (jyn514) [inline templates/releases/releases.html:57]: Why did you remove these HTML tags?

File: templates/core/home.html (hunk 1 above)
Comment (GuillaumeGomez) [reply templates/core/home.html:69]: We cannot focus on a `<li>`, so I removed the `<ul>` and moved the `<a>` out.

File: templates/releases/releases.html (hunk 2 above)
Comment (GuillaumeGomez) [reply templates/releases/releases.html:57]: We cannot focus on a `<li>`, so I removed the `<ul>` and moved the `<a>` out.

File: templates/core/home.html (hunk 1 above)
Comment (jyn514) [reply templates/core/home.html:69]: This seems like it might have the same accessibility problems as in https://github.com/rust-lang/rust/issues/87059 - is there another way you can do this?

Comment (GuillaumeGomez) [reply templates/core/home.html:69]: Hum... I don't know enough about accessibility to answer this. As for another way, I can focus on the `<a>` instead. So if you prefer for me to go this way, I'll update the PR to not change the DOM (or at least not change the current hierarchy).

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by GuillaumeGomez** [comment]
cc @Nemo157

**Comment by jyn514** [comment]
> The keyboard events are suboptimal currently. So I improved them a bit by following what jsha did in rustdoc.

Can you expand on this? What does this PR actually do?

**Comment by GuillaumeGomez** [comment]
Instead of simply highlighting an item with a CSS class, we focus it. Meaning that if you press enter, you go to the URL (and if you press ctrl+enter it opens it in another tab).

The other change is that pressing 's' is now focusing on the search input of the page (like in rustdoc). Just to be clear: this js isn't on the rustdoc pages so there is no "conflict" between the two in any way. It's on the search pages and the home page of docs.rs.

**Comment by jyn514** [comment]
@GuillaumeGomez this breaks autocomplete for the search box unfortunately:

https://user-images.githubusercontent.com/23638587/130337392-03d38630-07b6-4d0a-8e21-9d277bc418d3.mp4

**Comment by GuillaumeGomez** [comment]
Just like on rustdoc. Question then: should I disable it on the home page then or should I disable the autocomplete?

**Comment by jyn514** [comment]
Hmm, it seems unfortunate this is broken for rustdoc to.

@Manishearth @jsha do one of you have opinions on what to do hear? I kind of lean towards keeping the browser's default behavior.

**Comment by Manishearth** [comment]
Strongly lean towards not overriding browser default behavior

**Comment by GuillaumeGomez** [comment]
I'll disable this functionality on the home page then.

**Comment by GuillaumeGomez** [comment]
I disabled the usage of arrows on the keyboard (leaving only the escape key).

So that means:

 * If that if nothing is selected and you press up/down arrow, it'll go through the list.
 * If you press ESC, it'll blur the current active element
 * If you press 'S' and you're not in the search input, it'll focus on it.

**Comment by GuillaumeGomez** [comment]
Does it look good to everyone?

**Comment by GuillaumeGomez** [comment]
Ping?

**Comment by GuillaumeGomez** [comment]
Considering it's an improvement as it's centralize and make events handling coherent and that all comments were resolved, I'll merge it for the time being then. :)

**Comment by jyn514** [comment]
@GuillaumeGomez can you please avoid merging PRs without review? It wouldn't have gone out until today anyway, and I don't like bypassing feedback. Now if I find something wrong I have to delay the deploy until this is reverted.
